### ISSUE TITLE
Bean validation's @Size does not work with collections

### ISSUE THREAD

**Issue by magx2** [j-easy/easy-random#348]
I see that you do not support annotation `javax.validation.constraints.Size`. Can you support it?

From java doc:

> The annotated element size must be between the specified boundaries (included).
> Supported types are:
> CharSequence (length of character sequence is evaluated)
> Collection (collection size is evaluated)
> Map (map size is evaluated)
> Array (array length is evaluated)
> null elements are considered valid.

**Comment by benas** [comment]
`@Size` is supported and here is an example: [Annotated bean](https://github.com/benas/random-beans/blob/master/random-beans-validation/src/test/java/io/github/benas/randombeans/validation/BeanValidationAnnotatedBean.java#L97-L98) and [test](https://github.com/benas/random-beans/blob/master/random-beans-validation/src/test/java/io/github/benas/randombeans/validation/BeanValidationTest.java#L105).

> I see that you do not support annotation javax.validation.constraints.Size

Where do you see that?

**Comment by magx2** [comment]
Hmm I used it on `List` and it didn't work.

**Comment by benas** [comment]
It looks like [only the `String` type is supported](https://github.com/benas/random-beans/blob/master/random-beans-validation/src/main/java/io/github/benas/randombeans/validation/SizeAnnotationHandler.java#L54-L56). Here is a failing test with v3.9.0:

```java
import java.util.List;

import javax.validation.constraints.Size;

import io.github.benas.randombeans.EnhancedRandomBuilder;
import io.github.benas.randombeans.api.EnhancedRandom;
import org.junit.Assert;
import org.junit.Test;

public class Issue348 {

	@Test
	public void testSizeOnList() {
		// given
		EnhancedRandom enhancedRandom = new EnhancedRandomBuilder()
				.build();

		// when
		Person person = enhancedRandom.nextObject(Person.class);

		// then
		Assert.assertNotNull(person);
		int size = person.getNames().size();
		Assert.assertTrue(size >= 2 && size <= 5);
	}

	static class Person {
		@Size(min = 2, max = 5)
		private List<String> names;

		public Person() {
		}

		public List<String> getNames() {
			return names;
		}

		public void setNames(List<String> names) {
			this.names = names;
		}
	}
}
```

We need to add support for all types as documented in the annotation (`Collection`, `Map` and `Array`) in addition to Strings.

Thank you for reporting this issue!

**Comment by magx2** [comment]
Maybe use `Collection` interface instead of `List`?

**Comment by benas** [comment]
Yes as I said here:

> We need to add support for all types as documented in the annotation (`Collection`, `Map` and `Array`) in addition to Strings.

**Comment by benas** [comment]
@magx2 A fix has been deployed in version `4.0.0.RC2-SNAPSHOT`. Can you please give it a try?

If you don't know how to use a snapshot version, please refer to the [wiki here](https://github.com/j-easy/easy-random/wiki/Getting-Started#maven-usage).

As you might have noticed, the project has been renamed and you would need to adjust a couple of things (see [migration guide](https://github.com/j-easy/easy-random/releases/tag/easy-random-4.0.0.RC1)).

Looking forward for your feedback. Thank you upfront!

**Comment by magx2** [comment]
It's working! Thanks!
